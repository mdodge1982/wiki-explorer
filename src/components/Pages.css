.Pages {
	display: none;
	background: #fff;
	margin: 0;
	padding: 0;
	list-style-type: none;
}
.open .Pages {
	display: block;
}
