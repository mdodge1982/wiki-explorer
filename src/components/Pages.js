import React, {Component} from 'react';
import './Pages.css';
import Page from './Page';

class Pages extends Component {
	render() {
		return (
			<ul className="Pages">
				{this.props.pages.map(page => {
					return (<Page key={page.pageid} page={page}/>)
				})}
			</ul>
		);
	}
}

export default Pages;
