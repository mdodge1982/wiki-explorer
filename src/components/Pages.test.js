import React from 'react';
import Pages from './Pages';
import {Provider} from 'react-redux';
import store from '../store';
import renderer from 'react-test-renderer';

jest.mock('../actions/websocket');

test('Test render', () => {
	const pages = [
		{
			pageid: 1
		}, {
			pageid: 2
		}, {
			pageid: 3
		}
	];
	const component = renderer.create(
		<Provider store={store}>
			<Pages pages={pages}/>
		</Provider>
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();

});
