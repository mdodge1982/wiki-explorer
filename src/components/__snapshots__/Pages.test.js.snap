// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test render 1`] = `
<ul
  className="Pages"
>
  <li
    className="Page"
  >
    <h5
      onClick={[Function]}
    />
  </li>
  <li
    className="Page"
  >
    <h5
      onClick={[Function]}
    />
  </li>
  <li
    className="Page"
  >
    <h5
      onClick={[Function]}
    />
  </li>
</ul>
`;
